src/components/slider/slider.tsx

```diff
@@ -0,0 +1,433 @@
+                    numberOfLines={1}
+                    multiline={false}
+                  />
+                </View>
+              )}
+            </View>
+          )}
+
+          <View style={[styles.horizontalTrack, disabled && styles.disabledTrack]}>
+            <Animated.View
+              style={[
+                styles.filledTrack,
+                {
+                  left: isRange ? rangeLeftPos : 0,
+                  width: isRange
+                    ? Animated.subtract(rangeRightPos, rangeLeftPos)
+                    : Animated.add(handlePosition, new Animated.Value(handleSize / 2)),
+                },
+                disabled && styles.disabledTrack,
+              ]}
+            />
+
+            {internalMarkers.map((mark, index) => {
+              const markerLeft = ((mark - lowerLimit) / (upperLimit - lowerLimit)) * trackLength;
+              return (
+                <View key={index} style={[styles.internalMarker, { left: markerLeft }]}>
+                  <Text style={[styles.markerText, disabled && styles.disabledMarkers]}>
+                    {mark}
+                  </Text>
+                </View>
+              );
+            })}
+
+            {renderBubble(BubbleHandle.SINGLE, handlePosition, bubbleValue)}
+            {renderBubble(BubbleHandle.LEFT, rangeLeftPos, bubbleValue)}
+            {renderBubble(BubbleHandle.RIGHT, rangeRightPos, bubbleValue)}
+
+            {isRange ? (
+              <>
+                <Animated.View
+                  style={[
+                    styles.handle,
+                    disabled && styles.disabledHandle,
+                    { left: Animated.subtract(rangeLeftPos, handleSize / 2) },
+                  ]}
+                  {...leftPanResponder.panHandlers}
+                />
+                <Animated.View
+                  style={[
+                    styles.handle,
+                    disabled && styles.disabledHandle,
+                    { left: Animated.subtract(rangeRightPos, handleSize / 2) },
+                  ]}
+                  {...rightPanResponder.panHandlers}
+                />
+              </>
+            ) : (
+              <Animated.View
+                style={[
+                  styles.handle,
+                  disabled && styles.disabledHandle,
+                  { left: Animated.subtract(handlePosition, handleSize / 2) },
+                ]}
+                {...singlePanResponder.panHandlers}
+              />
+            )}
+          </View>
+
+          {showEndMarkers && (
+            <View style={styles.endMarkersRow}>
+              <Text style={[styles.markerText, disabled && styles.disabledMarkers]}>
+                {lowerLimit}
+              </Text>
+              <Text style={[styles.markerText, disabled && styles.disabledMarkers]}>
+                {upperLimit}
+              </Text>
+            </View>
+          )}
+        </View>
+      </View>
+    </View>
+  );
+};
+
+export default Slider;
```

Comment:
A file with 433 lines of code is extermely hard to reason and maintain.

Kindly find a way to refactor into smaller layouts, components, extend number or other types in TS and add appropriate functionality on them, and so forth.